client/value.go

```diff
@@ -89,3 +91,84 @@ func convertImmutable[T any](vals []immutable.Option[T]) []any {
 	}
 	return out
 }
+
+func (val *FieldValue) Kind() FieldKind {
+	return val.kind
+}
+
+func valOrNil[T any](value any) (*T, bool) {
```

Comment:
question: This function and the ones below appear to have limited usefulness within a public part of our codebase, why are they defined here, and not just done where are consumed?

Reply:
> But it seems like we want to get rid of CBOR.  Then the need for these methods will be more explicit.

How do you picture that being related?  The type of serialization used should not affect these getter funcs no?

thought: Now you have removed the nil stuff, further strangeness (to me) has been exposed - all these functions seem to do is convert the boolean `ok` of `v, ok := val.value.(T)` into an error, and I suspect most people prefer the boolean over an error.

Reply:
refactored